apps/emqx_management/src/emqx_mgmt_data_backup.erl

```diff
@@ -0,0 +1,674 @@
+        true ->
+            {ok, RawConf} = hocon:files([HoconFileName]),
+            {ok, _} = validate_cluster_hocon(RawConf),
+            maybe_print("Importing cluster configuration...~n", [], Opts),
+            %% At this point, when all validations have been passed, we want to log errors (if any)
+            %% but proceed with the next items, instead of aborting the whole import operation
+            do_import_conf(RawConf, Opts);
+        false ->
+            maybe_print("No cluster configuration to be imported.~n", [], Opts),
+            ?SLOG(info, #{
+                msg => "no_backup_hocon_config_to_import",
+                backup => BackupDir
+            }),
+            #{}
+    end.
+
+read_data_files(RawConf) ->
+    DataDir = bin(emqx:data_dir()),
+    RawConf1 = emqx_authz:maybe_read_acl_file(RawConf),
+    emqx_utils_maps:deep_convert(RawConf1, fun read_data_file/3, [DataDir]).
+
+read_data_file(Key, Val, DataDir) ->
+    Val1 =
+        case Val of
+            <<DataDir:(byte_size(DataDir))/binary, _RestPath/binary>> = FileName ->
+                case file:read_file(FileName) of
+                    {ok, Content} ->
+                        Content;
+                    {error, Reason} ->
+                        ?SLOG(warning, #{
+                            msg => "failed_to_read_data_file",
+                            filename => FileName,
+                            reason => Reason
+                        }),
+                        FileName
+                end;
+            V ->
+                V
+        end,
+    {Key, Val1}.
+
+validate_cluster_hocon(RawConf) ->
+    %% write ACL file to comply with the schema...
+    RawConf1 = emqx_authz:maybe_write_acl_file(RawConf),
+    emqx_hocon:check(
+        emqx_conf:schema_module(),
+        maps:merge(emqx:get_raw_config([]), RawConf1),
+        #{atom_key => false, required => false}
+    ).
+
+%% TODO: think about tombstone records (currently used in listeners only).
+%% Perhaps we should delete a listener if a tombstone is being imported.
+do_import_conf(RawConf, Opts) ->
+    GenConfErrs = filter_errors(maps:from_list(import_generic_conf(RawConf))),
+    maybe_print_errors(GenConfErrs, Opts),
+    lists:foldr(
+        fun(Module, ErrorsAcc) ->
+            {ok, #{changed := Changed, results := Results}} = Module:import_data(RawConf),
+            Errors = filter_errors(Results),
+            maybe_print_errors(Errors, Opts),
+            maybe_print_changed(Changed, Errors, Opts),
+            maps:merge(ErrorsAcc, Errors)
+        end,
+        GenConfErrs,
+        find_behaviours(emqx_config_backup)
+    ).
+
+import_generic_conf(Data) ->
+    lists:map(
+        fun(Key) ->
+            case maps:get(Key, Data, undefined) of
+                undefined -> {Key, ok};
+                Conf -> {Key, emqx_conf:update([Key], Conf, #{override_to => cluster})}
+            end
+        end,
+        ?CONF_KEYS
```

Comment:
Alternatively, we can try to import all keys present in cluster.hocon, skipping some keys that should never be imported (for example license)

Reply:
Agreed to keep explicit list of root keys on a review meeting.